src/index.ts

```diff
@@ -146,7 +146,7 @@ function fillOrders(side: string, price: number, quantity: number, userId: strin
       }
       if (asks[i].quantity > remainingQuantity) {
         asks[i].quantity -= remainingQuantity;
-        flipBalance(asks[i].userId, userId, remainingQuantity, price);
+        flipBalance(asks[i].userId, userId, remainingQuantity, asks[i].price);
```

Comment:
This isn't the only place this change needs to happen

Reply:
ok I will look deep into the code and try to find the affected part of the code :)